autogpt/utils.py

```diff
@@ -4,21 +4,23 @@
 import yaml
 from colorama import Fore
 from git.repo import Repo
+from prompt_toolkit import PromptSession
+from prompt_toolkit.history import InMemoryHistory
+from prompt_toolkit.formatted_text import FormattedText
+from typing import Union
 
-# Use readline if available (for clean_input)
-try:
-    import readline
-except:
-    pass
+session = PromptSession(history=InMemoryHistory())
 
-
-def clean_input(prompt: str = ""):
+def clean_input(prompt: Union[str, FormattedText] = ""):
```

Comment:
Is this used elsewhere in the repo?

Reply:
yeah clean_input is used in 5 places they are all covered